src/scenic/syntax/compiler.py

```diff
@@ -0,0 +1,1756 @@
+        "True if the transformer is processing a `compose` block of modular scenario"
+
+        self.inTryInterrupt = False
+        self.inInterruptBlock = False
+        self.inLoop = False
+        self.usedBreak = False
+        self.usedContinue = False
+
+    @property
+    def topLevel(self):
+        return (
+            not self.inBehavior
+            and not self.inMonitor
+            and not self.inCompose
+            and not self.inTryInterrupt
+        )
+
+    def context(
+        allowedContext: Context,
+        errorBuilder: Optional[Callable[[str], str]] = None,
+    ):
+        "Mark AST node as only available inside certain contexts"
+
+        def decorator(
+            visitor: Callable[["ScenicToPythonTransformer", ast.AST], ast.AST]
+        ):
+            def check_and_visit(self: "ScenicToPythonTransformer", node: ast.AST):
+                ctx = None
+                if self.topLevel and Context.TOP_LEVEL not in allowedContext:
+                    ctx = "at the top level"
+                elif self.inBehavior and Context.BEHAVIOR not in allowedContext:
+                    ctx = "inside a behavior"
+                elif self.inMonitor and Context.MONITOR not in allowedContext:
+                    ctx = "inside a monitor"
+                elif self.inCompose and Context.COMPOSE not in allowedContext:
+                    ctx = "inside a compose block"
+                if ctx:
+                    raise self.makeSyntaxError(
+                        f'Cannot use "{node.__class__.__name__}" {ctx}'
+                        if errorBuilder is None
+                        else errorBuilder(ctx),
+                        node,
+                    )
+                return visitor(self, node)
+
+            return check_and_visit
+
+        return decorator
+
+    def generic_visit(self, node):
+        if isinstance(node, s.AST):
+            raise Exception(
+                f'Scenic AST node "{node.__class__.__name__}" needs visitor in compiler'
+            )
+        return super().generic_visit(node)
+
+    # add support for list of nodes
+    def visit(self, node: Union[ast.AST, List[ast.AST]]):
+        if isinstance(node, ast.AST):
+            return super().visit(node)
+        elif isinstance(node, list):
+            newStatements = []
+            for statement in node:
+                newStatement = self.visit(statement)
+                if isinstance(newStatement, ast.AST):
+                    newStatements.append(newStatement)
+                else:
+                    newStatements.extend(newStatement)
+            return newStatements
+        else:
+            raise RuntimeError(f"unknown object {node} encountered during compilation")
+
+    # helper functions
+    def _register_requirement_syntax(self, syntax: ast.AST) -> int:
+        self.requirements.append(syntax)
+        return len(self.requirements) - 1
+
+    def visit_Name(self, node: ast.Name) -> any:
```

Comment:
@shumbo The return type annotation should be `Any` (you'll have to import it from `typing`), not the function `any`. There are a few more instances of this.